[feature-request] Keep error summary together at bottom of error list

When I run `ruff` with a subset of fixable errors, I get output like this:

```
Found 62 error(s).
dagster-cloud/integration_tests/python_modules/dagster-cloud-test-infra/dagster_cloud_test_infra/pex/sample-repos/repo-with-partition-set/repo1/repo1/__init__.py:1:25: F401 `repository.repo1` imported but unused
dagster-cloud/integration_tests/test_suites/kubernetes_dagster_cloud/test_kubernetes.py:137:31: F811 Redefinition of unused `dagster_cloud_most_recent_release` from line 31
... 60 more lines ...
11 potentially fixable with the --fix option.
```

IMO it's a bit confusing to have the number of fixable errors at the bottom and the total at the top (especially if there are a lot of errors and you have to scroll to the get the total)-- they should both be at the bottom.

--- a/src/printer.rs
+++ b/src/printer.rs
@@ -44,7 +44,7 @@ impl<'a> Printer<'a> {
         }
     }
 
-    fn pre_text(&self, diagnostics: &Diagnostics) {
+    fn post_text(&self, diagnostics: &Diagnostics, autofix: fixer::Mode) {
         if self.log_level >= &LogLevel::Default {
             let fixed = diagnostics.fixed;
             let remaining = diagnostics.messages.len();
@@ -54,13 +54,16 @@ impl<'a> Printer<'a> {
             } else if remaining > 0 {
                 println!("Found {remaining} error(s).");
             }
-        }
-    }
 
-    fn post_text(&self, num_fixable: usize, autofix: fixer::Mode) {
-        if self.log_level >= &LogLevel::Default {
-            if num_fixable > 0 && !matches!(autofix, fixer::Mode::Apply) {
-                println!("{num_fixable} potentially fixable with the --fix option.");
+            if !matches!(autofix, fixer::Mode::Apply) {
+                let num_fixable = diagnostics
+                    .messages
+                    .iter()
+                    .filter(|message| message.kind.fixable())
+                    .count();
+                if num_fixable > 0 {
+                    println!("{num_fixable} potentially fixable with the --fix option.");
+                }
             }
         }
     }
@@ -69,12 +72,6 @@ impl<'a> Printer<'a> {
         if matches!(self.log_level, LogLevel::Silent) {
             return Ok(());
         }
-
-        let num_fixable = diagnostics
-            .messages
-            .iter()
-            .filter(|message| message.kind.fixable())
-            .count();
 
         match self.format {
             SerializationFormat::Json => {
@@ -141,18 +138,13 @@ impl<'a> Printer<'a> {
                 println!("{}", report.to_string().unwrap());
             }
             SerializationFormat::Text => {
-                self.pre_text(diagnostics);
-
                 for message in &diagnostics.messages {
                     print_message(message);
                 }
 
-                self.post_text(num_fixable, autofix);
+                self.post_text(diagnostics, autofix);
             }
             SerializationFormat::Grouped => {
-                self.pre_text(diagnostics);
-                println!();
-
                 // Group by filename.
                 let mut grouped_messages = BTreeMap::default();
                 for message in &diagnostics.messages {
@@ -190,11 +182,9 @@ impl<'a> Printer<'a> {
                     println!();
                 }
 
-                self.post_text(num_fixable, autofix);
+                self.post_text(diagnostics, autofix);
             }
             SerializationFormat::Github => {
-                self.pre_text(diagnostics);
-
                 // Generate error workflow command in GitHub Actions format
                 // https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-error-message
                 diagnostics.messages.iter().for_each(|message| {

